pkg/image/definition.go

```diff
@@ -53,6 +53,9 @@ func ParseDefinition(data []byte) (*Definition, error) {
 	if err := yaml.Unmarshal(data, &definition); err != nil {
 		return nil, fmt.Errorf("could not parse the image definition: %w", err)
 	}
-
+	err := validate(&definition)
```

Comment:
I'd extract this out of here and instead export a `ValidateDefinition` function. This way we separate the functionality and concerns.